docs/resources/grant.md

```diff
@@ -0,0 +1,329 @@
+  name = aws_iam_role.external_data_access.name
+  aws_iam_role {
+    role_arn = aws_iam_role.external_data_access.arn
+  }
+  comment = "Managed by TF"
+}
+
+resource "databricks_grant" "external_creds" {
+  storage_credential = databricks_storage_credential.external.id
+
+  principal  = "Data Engineers"
+  privileges = ["CREATE_EXTERNAL_TABLE"]
+}
+```
+
+## External location grants
+
+You can grant `ALL_PRIVILEGES`, `CREATE_EXTERNAL_TABLE`, `CREATE_MANAGED_STORAGE`, `CREATE EXTERNAL VOLUME`, `READ_FILES` and `WRITE_FILES` privileges to [databricks_external_location](external_location.md) id specified in `external_location` attribute:
+
+```hcl
+resource "databricks_external_location" "some" {
+  name            = "external"
+  url             = "s3://${aws_s3_bucket.external.id}/some"
+  credential_name = databricks_storage_credential.external.id
+  comment         = "Managed by TF"
+}
+
+resource "databricks_grant" "some_data_engineers" {
+  external_location = databricks_external_location.some.id
+
+  principal  = "Data Engineers"
+  privileges = ["CREATE_EXTERNAL_TABLE", "READ_FILES"]
+}
+
+resource "databricks_grant" "some_service_principal" {
+  external_location = databricks_external_location.some.id
+
+  principal  = databricks_service_principal.my_sp.application_id
+  privileges = ["USE_SCHEMA", "MODIFY"]
+}
+
+resource "databricks_grant" "some_group" {
+  external_location = databricks_external_location.some.id
+
+  principal  = databricks_group.my_group.display_name
+  privileges = ["USE_SCHEMA", "MODIFY"]
+}
+
+resource "databricks_grant" "some_user" {
+  external_location = databricks_external_location.some.id
+
+  principal  = databricks_group.my_user.user_name
+  privileges = ["USE_SCHEMA", "MODIFY"]
+}
+```
+
+## Connection grants
+
+You can grant `ALL_PRIVILEGES`, `USE_CONNECTION` and `CREATE_FOREIGN_CATALOG` to [databricks_connection](connection.md) specified in `foreign_connection` attribute:
+
+```hcl
+resource "databricks_connection" "mysql" {
+  name            = "mysql_connection"
+  connection_type = "MYSQL"
+  comment         = "this is a connection to mysql db"
+  options = {
+    host     = "test.mysql.database.azure.com"
+    port     = "3306"
+    user     = "user"
+    password = "password"
+  }
+  properties = {
+    purpose = "testing"
+  }
+}
+
+resource "databricks_grant" "some" {
+  foreign_connection = databricks_connection.mysql.name
+
+  principal  = "Data Engineers"
+  privileges = ["CREATE_FOREIGN_CATALOG", "USE_CONNECTION"]
+}
+```
+
+## Delta Sharing share grants
+
+You can grant `SELECT` to [databricks_recipient](recipient.md) on [databricks_share](share.md) name specified in `share` attribute:
+
+```hcl
+resource "databricks_share" "some" {
+  name = "my_share"
+}
+
+resource "databricks_recipient" "some" {
+  name = "my_recipient"
+}
+
+resource "databricks_grant" "some" {
+  share = databricks_share.some.name
+
+  principal  = databricks_recipient.some.name
+  privileges = ["SELECT"]
+}
+```
```

Comment:
I was suggesting to leave the HCL examples here, but defer to grants.md on what permissions can be applied to a resource (as future grants will be added)

Reply:
Done